app/helpers/compute_resources_vms_helper.rb

```diff
@@ -263,23 +263,24 @@ def vm_import_action(vm, html_options = {})
         :type => 'unmanaged'),
       html_options
     )
-
     import_managed_link + import_unmanaged_link
   end
 
-  def vm_associate_action(vm)
-    display_link_if_authorized(
-      _("Associate VM"),
-      hash_for_associate_compute_resource_vm_path(
-        :compute_resource_id => @compute_resource,
-        :id => vm.identity
-      ).merge(
-        :auth_object => @compute_resource,
-        :permission => 'edit_compute_resources'),
-      :title => _("Associate VM to a Foreman host"),
-      :method => :put,
-      :class => "btn btn-default"
-    )
+  def vm_associate_action(vm, isdropdown: false)
+    if @compute_resource.supports_host_association?
```

Comment:
```suggestion
  return unless @compute_resource.supports_host_association?
```

A guard clause can reduce the level of indentation.